Library/Homebrew/dev-cmd/bottle.rb

```diff
@@ -266,39 +275,92 @@ def bottle_formula(f, args:)
       return
     end
 
-    return ofail "Formula was not installed with --build-bottle: #{f.full_name}" unless Utils::Bottles.built_as? f
-
     return ofail "Formula has no stable version: #{f.full_name}" unless f.stable
 
-    if args.no_rebuild? || !f.tap
-      rebuild = 0
+    bottle_tag, rebuild = if local_bottle_json
```

Comment:
Shouldn't `local_bottle_json` always evaluate to true by the time we get here? Given the `return ofail`s above.

Though you do have a pretty large `if`/`then` block later that conditions on this, so maybe I'm missing something.

Reply:
Ah, wait, I missed the `unless`es in the `return statements. Never mind.